### PR TITLE
feat(datepicker): indicate selected day in aria label

### PR DESCRIPTION
**Pull request by d-kostov-dev**
fixes VPAT-3637

PR Type
Feature - Accessibility

What is the current behavior?
The currently selected date is not indicated to screen reader users. 
 
What is the new behavior?
The aria-label of the day is now showing the word "Selected".

Does this PR introduce a breaking change?
No.

### CODE REVIEW

File: projects/angular/src/forms/datepicker/day.ts [hunk 1]
```diff
@@ -50,7 +51,7 @@ export class ClrDay {
   @Input('clrDayView')
   public set dayView(day: DayViewModel) {
     this._dayView = day;
-    this.dayString = this._dayView.dayModel.toDateString();
+    this.dayString = `${this._dayView.dayModel.toDateString()} ${this.dayView.isSelected ? this.commonStrings.keys.datepickerSelectedLabel : ''}`;
```
Comment (ashleyryan) [inline projects/angular/src/forms/datepicker/day.ts:54]: You can't just concatenate like this, since not every language uses this structure. You'll need to put a placeholder in the string.

I need to run this by our A11Y team before we can accept it

File: projects/angular/src/utils/i18n/common-strings.default.ts [hunk 2]
```diff
@@ -68,6 +68,7 @@ export const commonStringsDefault: ClrCommonStrings = {
   datepickerCurrentDecade: 'Current decade',
   datepickerSelectMonthText: 'Select month, the current month is {CALENDAR_MONTH}',
   datepickerSelectYearText: 'Select year, the current year is {CALENDAR_YEAR}',
+  datepickerSelectedLabel: 'Selected',
```
Comment (kevinbuhmann) [inline projects/angular/src/utils/i18n/common-strings.default.ts:71]: I think this should be consistent with `datepickerSelectMonthText` and `datepickerSelectMonthText`.

```typescript
datepickerSelectDayText: 'Select day, the current day is {CALENDER_DAY}`
```

**Review by kevinbuhmann — CHANGES_REQUESTED** [review]

File: projects/angular/src/forms/datepicker/day.ts (hunk 1 above)
Comment (steve-haar) [reply projects/angular/src/forms/datepicker/day.ts]: Technically, this will add a trailing space to the end of the day if it's not selected. I doubt it will affect screen readers, but if we wanted to be more correct we could do something like:

```
const selectedText = this.dayView.isSelected ? ` ${this.commonStrings.keys.datepickerSelectedLabel}` : '';
this.dayString = `${this._dayView.dayModel.toDateString()}${selectedText}`
```

Comment (kevinbuhmann) [reply projects/angular/src/forms/datepicker/day.ts]: We cannot use concatenation though, like Ashley said. We need a placeholder in the common string.

**Comment by kevinbuhmann** [comment]
I force pushed this branch to get the PR building for a11y review and to reword the commit message to follow our conventions.

**Comment by ashleyryan** [comment]
@d-kostov-dev Here is the recommendation from our A11Y expert:

> Because aria-selected is not widely supported in either td or button for screen readers we recommend using aria-label="[full date] - Selected" instead. Additionally aria-current="date" should also be provided to indicate the current date as Clarity's date picker does show in visually in bold what the current date is when it is not selected.

They included a picture too: 
image

**Review by kevinbuhmann — CHANGES_REQUESTED** [review]
See comment. Also, please fix the build errors (including lint and test errors).

File: projects/angular/src/forms/datepicker/day.ts [hunk 3]
```diff
@@ -27,6 +27,7 @@ import { DateNavigationService } from './providers/date-navigation.service';
       (click)="selectDay()"
       (focus)="onDayViewFocus()"
       [attr.aria-label]="dayString"
+      [attr.aria-current]="dayView.isTodaysDate"
```
Comment (kevinbuhmann) [inline projects/angular/src/forms/datepicker/day.ts:30]: This isn't quite right because the value should be literally "date" as shown in the graphic. But this is actually unrelated to I indicating the selected day, so I think this should removed. I'll open a separate PR.

**Review by kevinbuhmann — CHANGES_REQUESTED** [review]
See comment.

Also, please squash the commits into one "feat(datepicker): indicate selected day in aria-label" commit.

**Review by kevinbuhmann — CHANGES_REQUESTED** [review]
Please also run `npm run build && npm run public-api:update` to update the public API file.

File: projects/angular/src/forms/datepicker/day.ts (hunk 3 above)
Comment (kevinbuhmann) [reply projects/angular/src/forms/datepicker/day.ts:30]: This is in #253.

**Review by kevinbuhmann — CHANGES_REQUESTED** [review]
Also, please add a test for this change. You can use this if you want (after the "sets the correct value for the aria-label attribute" test):

```typescript
it('sets the correct value for the aria-label attribute when the date is selected', () => {
  const dayBtn: HTMLButtonElement = context.clarityElement.children[0];
  const dvm: DayViewModel = context.clarityDirective.dayView;

  context.testComponent.dayView.isSelected = true;

  context.detectChanges();
  expect(dayBtn.attributes['aria-label'].value).toEqual(`${dvm.dayModel.toDateString()} - SELECTED`);
});
```

File: projects/angular/src/forms/datepicker/day.ts [hunk 4]
```diff
@@ -50,13 +52,20 @@ export class ClrDay {
...
     this._dayView = day;
-    this.dayString = this._dayView.dayModel.toDateString();
   }
 
   public get dayView(): DayViewModel {
     return this._dayView;
   }
 
+  public get dayString(): string {
+    return this.dayView.isSelected
```
Comment (kevinbuhmann) [inline projects/angular/src/forms/datepicker/day.ts:62]: Why was this moved?

Comment (d-kostov-dev) [reply projects/angular/src/forms/datepicker/day.ts:62]: The dayString is initialized with the input. And it works ok in the app because every time a date is selected the grid closes and when you open it, it initializes again, but the change detection on the unit test is not working this way and the test is not passing, that's why I moved it to getter. Maybe should be private or there is an other solution to this.

Comment (kevinbuhmann) [reply projects/angular/src/forms/datepicker/day.ts:62]: Ah, okay.

**Review by kevinbuhmann — CHANGES_REQUESTED** [review]
Please squash the commits and keep the first commit message. These commits are not compatible with our release process.

Also, please rebase and run `npm run build && npm run public-api:update` to fix the build error.

File: projects/angular/src/forms/datepicker/day.ts [hunk 5]
```diff
@@ -25,7 +26,9 @@ import { DateNavigationService } from './providers/date-navigation.service';
       [attr.tabindex]="dayView.tabIndex"
       (click)="selectDay()"
       (focus)="onDayViewFocus()"
+      [attr.aria-current]="dayView.isTodaysDate ? 'date' : 'false'"
```
Comment (kevinbuhmann) [inline projects/angular/src/forms/datepicker/day.ts:29]: This should not be here. Please fetch the upstream repo and rebase.

**Review by kevinbuhmann — CHANGES_REQUESTED** [review]

**Comment by kevinbuhmann** [comment]
Sorry if you are still working on the rebase, but I see the merge commit you just pushed. We don't want a merge commit. There should be just one commit without any of the `aria-current` stuff. I have already merged via a different PR.

File: projects/angular/src/forms/datepicker/day.spec.ts [hunk 6]
```diff
@@ -122,12 +122,36 @@ export default function () {
         expect(context.clarityDirective.selectDay).toHaveBeenCalled();
       });
 
+      it('sets arria-current on the button correctly', function () {
```
Comment (kevinbuhmann) [inline projects/angular/src/forms/datepicker/day.spec.ts:125]: This should be removed.

**Comment by kevinbuhmann** [comment]
I went ahead and rebased for you to remove the `aria-current` changes from this commit. When the build passes, I will merge.

**Review by ashleyryan — APPROVED** [review]

**Review by kevinbuhmann — APPROVED** [review]
